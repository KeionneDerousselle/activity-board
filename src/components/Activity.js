import React from 'react';
import PropTypes from 'prop-types';

const Activity = ({...activity}) => 
  <div>
    <div>{activity.title}</div>
    {activity.description || <div>{activity.description}</div> }
    <div>{activity.type}</div>
    <div>{activity.rating}</div>
    <div>{activity.price}</div>
  </div>;

Activity.propTypes = {
  title: PropTypes.string.isRequired,
  description: PropTypes.string,
  rating: PropTypes.number,
  type: PropTypes.string.isRequired,
  tags: PropTypes.array,
  price: PropTypes.number
};

export default Activity;
