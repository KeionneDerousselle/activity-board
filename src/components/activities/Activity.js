import React from 'react';
import PropTypes from 'prop-types';
import { css } from 'react-emotion';
import { connect } from 'react-redux';
import { Row, Col } from 'antd';
import { MainLayout } from '../layout';
import ActivityDetails from './ActivityDetails';

const mdSizing = {
  span: 16,
  offset: 4
};

const smSizing = {
  span: 22,
  offset: 1
};

const headerClass = css({
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center'
});

class Activity extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      activity: { ...props.activity },
      activityIsFetching: props.activityIsFetching
    };
  }

  componentWillReceiveProps(nextProps) {
    if (this.props.activity !== nextProps.activity) {
      this.setState({ activity: { ...nextProps.activity } });
    }
    if (this.props.activityIsFetching !== nextProps.activityIsFetching) {
      this.setState({ activityIsFetching: nextProps.activityIsFetching });
    }
  }

  render() {
    const { activity, activityIsFetching } = this.state;
    const { tags } = this.props;

    const tagsObj = tags.items.reduce((result, item) => {
      result[item.id] = item.name;
      return result;
    }, {});

    return (
      <MainLayout
        isContentLoading={!activity || activityIsFetching || !tags || tags.isFetching}
        content={
          <Row>
            <Col
              xs={smSizing}
              sm={smSizing}
              md={mdSizing}
              lg={mdSizing}
              xl={mdSizing}
              xxl={mdSizing}
            >
              <div>
                <div className={headerClass}>
                  <h1>{activity.title}</h1>
                </div>
                <ActivityDetails
                  tags={tagsObj}
                  img={activity.img}
                  price={activity.price}
                  description={activity.description}
                  rating={activity.rating}
                />
              </div>
            </Col>
          </Row>
        }
      />
    );
  }
}

Activity.propTypes = {
  activity: PropTypes.object,
  activityIsFetching: PropTypes.bool.isRequired,
  tags: PropTypes.object.isRequired,
  actions: PropTypes.shape({
    fetchTagsIfNeeded: PropTypes.func.isRequired
  }).isRequired
};

const getActivityById = (activities, id) =>
  activities.find(activity => activity.id == id);

const mapStateToProps = (state, ownProps) => {
  const activityId = ownProps.match.params.id;
  const activities = state.activities.items;
  const activity = getActivityById(activities, activityId);
  return {
    activityIsFetching: state.activities.isFetching,
    activity: activity,
    tags: state.tags
  };
};

export default connect(mapStateToProps)(Activity);
