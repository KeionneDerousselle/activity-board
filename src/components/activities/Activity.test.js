import Activity from './Activity';
import React from 'react';
import { shallow } from 'enzyme';

let wrapper;
const title = 'foo title';
const description = 'foo description';
const rating = 3.5;
const tags = ['foo-tag-1', 'foo-tag-2', 'foo-tag-3', 'foo-tag-4']
const price = 100;

beforeEach(()=>{
  wrapper = shallow(<Activity 
    title={title}
    description={description}
    rating={rating}
    tags={tags}
    price={price}
  />);
});

test('Activity Snapshot', ()=>{
  expect(wrapper).toMatchSnapshot();
});
