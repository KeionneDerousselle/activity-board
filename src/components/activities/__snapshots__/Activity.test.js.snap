// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Activity Snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <withRouter()
    description="foo description"
    price={100}
    rating={3.5}
    tags={
      Array [
        "foo-tag-1",
        "foo-tag-2",
        "foo-tag-3",
        "foo-tag-4",
      ]
    }
    title="foo title"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "render": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "render": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
